pkg/parcacol/sample.go

```diff
@@ -156,29 +157,32 @@ func SampleToParquetRow(
 
 		// All remaining cases take care of dynamic columns
 		case ColumnLabels:
-			for _, label := range ls {
-				row = append(row, parquet.ValueOf(label.Value).Level(0, 1, columnIndex))
+			lset := ls.Map() // Convert labels to map for quicker lookup in loop.
```

Comment:
this seems very expensive, maybe we can pass a map in the first place and let the caller decide how to optimize these allocations?